op-service/eth/blob.go

```diff
@@ -69,59 +71,198 @@ func VerifyBlobProof(blob *Blob, commitment kzg4844.Commitment, proof kzg4844.Pr
+	b[1] = EncodingVersion
+
+	// encode the length as big-endian uint24 into [2:5] bytes of the first field element
+	if len(data) < 1<<24 {
+		// Zero out any trailing data in the buffer if any
+		b[2] = byte((len(data) >> 16) & 0xFF) // Most significant byte
+		b[3] = byte((len(data) >> 8) & 0xFF)
+		b[4] = byte(len(data) & 0xFF) // Least significant byte
+	} else {
+		return fmt.Errorf("Error: length_rollup_data is too large")
+	}
+
+	offset := 0
+	var buffer []byte
+
+	// encode the first 27 + 1 bytes of data into remaining bytes of first field element
+	buffer, offset = read(0, 27, data)
+	x, offset := read(offset, 1, data)
+	encodedByte := x[0] & 0b0011_1111
+	b.write([]byte{encodedByte}, 0)
+	b.write(buffer, 5)
+	pointer := 32 // manually set the pointer to the next field element
+
+	clearBuffer(buffer)
+	buffer, offset = read(offset, 31, data)
+	y, offset := read(offset, 1, data)
+
+	encodedByte = (y[0] & 0b0000_1111) | ((x[0] & 0b1100_0000) >> 2)
+	pointer = b.write([]byte{encodedByte}, pointer)
+	pointer = b.write(buffer, pointer)
+
+	clearBuffer(buffer)
+	buffer, offset = read(offset, 31, data)
+	z, offset := read(offset, 1, data)
+	encodedByte = z[0] & 0b0011_1111
+	pointer = b.write([]byte{encodedByte}, pointer)
+	pointer = b.write(buffer, pointer)
+
+	clearBuffer(buffer)
+	buffer, offset = read(offset, 31, data)
+	encodedByte = ((z[0] & 0b1100_0000) >> 2) | ((y[0] & 0b1111_0000) >> 4)
+	pointer = b.write([]byte{encodedByte}, pointer)
+	pointer = b.write(buffer, pointer)
+
+	if offset == len(data) {
+		return nil
+	}
+
+	for fieldNumber := 1; fieldNumber < 1024; fieldNumber++ {
+		clearBuffer(buffer)
+		buffer, offset = read(offset, 31, data)
+		x, offset = read(offset, 1, data)
+		encodedByte = x[0] & 0b0011_1111
+		pointer = b.write([]byte{encodedByte}, pointer)
+		pointer = b.write(buffer, pointer)
+
+		clearBuffer(buffer)
+		buffer, offset = read(offset, 31, data)
+		y, offset = read(offset, 1, data)
+		encodedByte = (y[0] & 0b0000_1111) | ((x[0] & 0b1100_0000) >> 2)
+		pointer = b.write([]byte{encodedByte}, pointer)
+		pointer = b.write(buffer, pointer)
+
+		clearBuffer(buffer)
+		buffer, offset = read(offset, 31, data)
+		z, offset = read(offset, 1, data)
+		encodedByte = z[0] & 0b0011_1111
+		pointer = b.write([]byte{encodedByte}, pointer)
+		pointer = b.write(buffer, pointer)
+
+		clearBuffer(buffer)
+		buffer, offset = read(offset, 31, data)
+		encodedByte = ((z[0] & 0b1100_0000) >> 2) | ((y[0] & 0b1111_0000) >> 4)
+		pointer = b.write([]byte{encodedByte}, pointer)
+		pointer = b.write(buffer, pointer)
+
+		if offset >= len(data) {
+			return nil
 		}
 	}
+
 	if offset < len(data) {
 		return fmt.Errorf("failed to fit all data into blob. bytes remaining: %v", len(data)-offset)
 	}
+
 	return nil
 }
 
+func read(offset int, numBytes int, data []byte) ([]byte, int) {
+	if offset >= len(data) {
+		// If the offset is at or beyond the end of data, return a new byte array of numBytes length
```

Comment:
cute, I was wondering how you were dealing with input overflow... I would put this detail about short reads in the godoc above the function as it's important to understanding how the encoding works.